avogadro/core/molecule.cpp

```diff
@@ -1324,4 +1324,53 @@ std::list<Index> Molecule::getAtomsAtLayer(size_t layer)
   return result;
 }
 
+void Molecule::boundingBox(Vector3& boxMin, Vector3& boxMax) const
+{
+  boxMin.setConstant(std::numeric_limits<double>::max());
+  boxMax.setConstant(-std::numeric_limits<double>::max());
+
+  bool noSelection = true;
+
+  for (uint32_t i = 0; i < m_selectedAtoms.size(); i++) {
+    if (m_selectedAtoms[i]) {
+      noSelection = false;
+      break;
+    }
+  }
+
+  for (uint32_t i = 0; i < atomCount(); i++) {
+    if (noSelection || m_selectedAtoms[i]) {
+      double radius = 1.0;
+
+      Vector3 boxMinBuffer;
+      Vector3 boxMaxBuffer;
+
+      boxMinBuffer.x() = atom(i).position3d().x() - radius;
+      boxMinBuffer.y() = atom(i).position3d().y() - radius;
+      boxMinBuffer.z() = atom(i).position3d().z() - radius;
+      boxMaxBuffer.x() = atom(i).position3d().x() + radius;
+      boxMaxBuffer.y() = atom(i).position3d().y() + radius;
+      boxMaxBuffer.z() = atom(i).position3d().z() + radius;
+
+      if (boxMinBuffer.x() < boxMin.x())
+        boxMin.x() = boxMinBuffer.x();
```

Comment:
boxMin.x() = min(boxMin.x(), boxMinBuffer.x());

etc. .. which is more readable, since the library handles the conditional